cmake/common.cmake

```diff
@@ -0,0 +1,35 @@
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin")
+    
+string(ASCII 27 Esc)
+set(Green    "${Esc}[92m"     CACHE STRING "green color"    FORCE)
+set(Blue     "${Esc}[1;94m"   CACHE STRING "Blue color"     FORCE)
+set(BoldRed  "${Esc}[1;5;91m" CACHE STRING "bold-red color" FORCE)
+set(NoColor  "${Esc}[0m"      CACHE STRING "reset color"    FORCE)
+
+if(NOT WIN32)
+
+    execute_process(COMMAND   ${CMAKE_C_COMPILER}  -dumpversion   OUTPUT_VARIABLE   GCC_VERSION)
+    if(GCC_VERSION VERSION_GREATER 4.9 OR GCC_VERSION VERSION_EQUAL 4.9)
+        set(STDLIBC  "-std=c++1y")
```

Comment:
"we don't use ..."  presumably that could change right? :-)  

 It's nice to be able to use bounds-checking structures with simple syntax. 
 For example: 
       `**std::vector<double> coefficients = {1.0, 2.0,  rand()*1.0 };** ` 
  looks a lot simpler than however you'd do it in C++03 (for-loop?), and safer than a raw C-style array. 

Also, on Windows, visual studio 2012 by default will enable C++11, while visual studio 2013 enables C++14. These gcc flags are to align the default compiler behavior across platforms.